private/bufpkg/bufimage/bufimagemodify/internal/location_path_dfa.go

```diff
@@ -0,0 +1,112 @@
+// Copyright 2020-2023 Buf Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package internal
+
+const (
+	// https://github.com/protocolbuffers/protobuf/blob/29152fbc064921ca982d64a3a9eae1daa8f979bb/src/google/protobuf/descriptor.proto#L75
+	messageTypeTagInFile int32 = 4
+	// https://github.com/protocolbuffers/protobuf/blob/29152fbc064921ca982d64a3a9eae1daa8f979bb/src/google/protobuf/descriptor.proto#L78
+	extensionTagInFile = 7
+	// https://github.com/protocolbuffers/protobuf/blob/29152fbc064921ca982d64a3a9eae1daa8f979bb/src/google/protobuf/descriptor.proto#L97
+	fieldTagInMessage = 2
+	// https://github.com/protocolbuffers/protobuf/blob/29152fbc064921ca982d64a3a9eae1daa8f979bb/src/google/protobuf/descriptor.proto#L100
+	nestedTypeTagInMessage = 3
+	// https://github.com/protocolbuffers/protobuf/blob/29152fbc064921ca982d64a3a9eae1daa8f979bb/src/google/protobuf/descriptor.proto#L98
+	extensionTagInMessage = 6
+	// https://github.com/protocolbuffers/protobuf/blob/29152fbc064921ca982d64a3a9eae1daa8f979bb/src/google/protobuf/descriptor.proto#L215
+	optionsTagInField          = 8
+	pathTypeInvalid   pathType = iota + 1
+	pathTypeEmpty
+	pathTypeMessages
+	pathTypeMessage
+	pathTypeFields
+	pathTypeField
+	pathTypeFieldOptions
+	pathTypeSomeFieldOption
+)
+
+type pathType int
+
+func isPathForFieldOptions(path []int32) bool {
+	return getPathType(path) == pathTypeFieldOptions
+}
+
+func isPathForSomeFieldOption(path []int32) bool {
+	return getPathType(path) == pathTypeSomeFieldOption
+}
```

Comment:
Related to previous comment: I don't think you actually need these helpers at all. The place that calls them can instead call `getPathType` and then do its own comparisons of the type to `pathTypeFieldOptions` or `pathTypeSomeFieldOption` so it doesn't need to run the automata more than once.